### DISCUSSION TITLE
Avalonia versioning with UserControl Nuget package

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#6872] [Q&A] [answered]
Hi,

Let's suppose I produce a UserControl linked against Avalonia 0.10 and release it as Nuget package.

Then, in a 6 months time, someone references my control in their Avalonia 0.20 project. How is the Avalonia versioning resolved? Which version does my UserControl control reference in their build?

Thanks in advance

**Comment by kekekeks** [comment]
We are using [Semantic Versioning](https://semver.org/) model. That's the only reason why the major version number is still zero.

**Comment by kuiperzone** [comment, marked as answer]
I perhaps did not formulate my question properly. I guess it is related to the issue of resolving package dependencies which I perhaps am a little new to.

By way of example, if we look at this existing control for Avalonia on Nuget: https://www.nuget.org/packages/Avalonia.Controls.PanAndZoom/

We can see it states:

  .NETStandard 2.0
  Avalonia (>= 0.10.10)
  
  .NETStandard 2.1
  Avalonia (>= 0.10.10)
  
  net5.0
  Avalonia (>= 0.10.10)

Presumably, if I am using a future Avalonia release, say 0.15.1 or something, I am fine to use a control linked against an older Avalonia version?

**Reply by kekekeks** [reply]
You are safe to use a control linked with 0.10.x in app linked with any newer _0.10.x_ release. Breaking changes do occur between major releases.

Also see:
https://github.com/AvaloniaUI/Avalonia/issues/2239#issuecomment-939438205

**Reply by kuiperzone** [reply]
Thank you. That's the answer.
